core/iwasm/compilation/aot_llvm.c

```diff
@@ -537,8 +539,51 @@ aot_build_precheck_function(AOTCompContext *comp_ctx, LLVMModuleRef module,
     if (ret_type == VOID_TYPE) {
         name = "";
     }
-    LLVMValueRef retval =
-        LLVMBuildCall2(b, func_type, wrapped_func, params, param_count, name);
+
+    LLVMValueRef retval;
+    if (comp_ctx->is_indirect_mode
+        && !strncmp(comp_ctx->target_arch, "xtensa", 6)) {
+        /* call wrapped_func indirectly */
+        if (!create_func_ptrs(comp_ctx, func_ctx)) {
```

Comment:
create_func_ptrs here may cause func_ptrs duplicated created since it will be created again in:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/169e1648156d008825831fb00d5e3acbe021b480/core/iwasm/compilation/aot_llvm.c#L1775-L1777

Could we remove the code here, and move the code in the above link to be before calling aot_add_llvm_func:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/169e1648156d008825831fb00d5e3acbe021b480/core/iwasm/compilation/aot_llvm.c#L1705-L1709

Reply:
These two func_ptrs are pointed to different func_ctx, one is for the original function context like aot_func_internal#0, and the other one is for precheck function like aot_func#0. 

Different func_ctx can not use the same func_ptr, if we try to use the func_ptr of a func_ctx in other func_ctx,  error log like "Referring to an argument in another function" will output during compilation. So it may be better to call create_func_ptrs() twice for two func_ctx.